torch/_higher_order_ops/triton_kernel_wrap.py

```diff
@@ -1091,12 +1091,22 @@ def init_variable(
                 and (
                     # pre_hook requires running arbitrary code at runtime, which we cannot handle at this time
                     # https://github.com/pytorch/pytorch/issues/139059
-                    # Check Config passed to autotuner in configs
-                    any(cfg.pre_hook is not None for cfg in kernel.configs)
+                    # we can't support pre_hook or post_hook in user defined triton kernels at the moment,
+                    # as they require the ability to execute code at runtime (AOTI can't support this)
+                    (
+                        "user_defined_pre_hook" in defaults
```

Comment:
`user_defined_pre_hook` is not a parameter of the `Autotuner.__init__`. Should we check it as the `kernel` attribute? I.e., `hasattr(kernel, "user_defined_pre_hook")`. Ditto the post-hook below.